providers/implementations/ciphers/ciphercommon_block.c

```diff
@@ -120,18 +128,19 @@ int ossl_cipher_unpadblock(unsigned char *buf, size_t *buflen, size_t blocksize)
      * Otherwise it provides a padding oracle.
      */
     pad = buf[blocksize - 1];
-    if (pad == 0 || pad > blocksize) {
-        ERR_raise(ERR_LIB_PROV, PROV_R_BAD_DECRYPT);
-        return 0;
-    }
-    for (i = 0; i < pad; i++) {
-        if (buf[--len] != pad) {
-            ERR_raise(ERR_LIB_PROV, PROV_R_BAD_DECRYPT);
-            return 0;
-        }
+    gp = ~(constant_time_is_zero_s(pad) | constant_time_lt_s(blocksize, pad));
```

Comment:
In general - I wish we left the original code inside  a comment :)

Reply:
Is this what you had in mind?